### ISSUE TITLE
Peek without peekable

### ISSUE THREAD

**Issue by themiurgo** [erikrose/more-itertools#53]
Hello,

sometimes I need to peek at the first element of an iterable. I usually implement it like this:

```
def peek(iterable):
    iterator = iter(iterable)
    item = next(iterator)
    return item, itertools.chain([item], iterator)

element, my_list = peek(my_list)
```


Would this be an interesting addition to more-itertools? If so, I'll send a pull request sometime in the future.

Cheers.

**Comment by mathieulongtin** [comment]
I managed to make peekable 3 times faster, but it's still way slower than this, and it fits a lot of my use cases for peekable. Thanks.

```
In [63]: data = range(10000)

In [64]: %timeit sum(peekable(iter(data)))
The slowest run took 4.38 times longer than the fastest. This could mean that an intermediate result is being cached
10 loops, best of 3: 26.6 ms per loop

In [65]: %timeit sum(peek(data)[1])
1000 loops, best of 3: 242 µs per loop

In [66]:
```

**Comment by themiurgo** [comment]
Glad it is useful! If you ever want it as a drop-in function and want to avoid rewriting it, it is included in the `toolz` library: https://github.com/pytoolz/toolz/pull/244

**Comment by mathieulongtin** [comment]
Didn't know about toolz, looks great. Thanks. 

I made slight modification to peek so that it can return a default instead of raise StopIteration:

```
_marker = object()
def peek(iterable, default=_marker):
      iterator = iter(iterable)
      try:
          item = next(iterator)
          return item, itertools.chain([item], iterator)
      except StopIteration:
          if default == _marker:
              raise
          return default, iterator
```

**Comment by bbayles** [comment]
I'm trying to decide on what to do with this issue. On the one hand: `toolz` already has this, `more_itertools` already has a serviceable-but-slower method of doing the same thing. On the other hand, this procedure is _way_ faster than the existing one.

I did find a 2x-3x speedup for the existing `peekable` class - I'll make a PR for that shortly. However, that's still much slower than this method.

I'm inclined to (a) rename this function from `peek` to `look_ahead`, and (b) extend it to return an arbitrary number of elements from the head of the iterator.

**Comment by erikrose** [comment]
I think your inclination is fine. Another fun (and short) name is `spy`.

**Comment by bbayles** [comment]
Fixed in #76 (and #76) - many thanks @themiurgo and @mathieulongtin.
